tap_impact/sync.py

```diff
@@ -143,38 +156,41 @@ def sync_endpoint(client,
     last_integer = None
     max_bookmark_value = None
 
+    end_dttm = utils.now()
+    end_dt = end_dttm.date()
+    start_dt = end_dt
+
     if bookmark_type == 'integer':
         last_integer = get_bookmark(state, stream_name, 0)
         max_bookmark_value = last_integer
     else:
         last_datetime = get_bookmark(state, stream_name, start_date)
+        last_datetime_dt = datetime.fromisoformat(last_datetime.replace('Z', '+00:00'))
+        default_date = end_dttm - timedelta(days=3*365)
+        default_date_str = default_date.strftime('%Y-%m-%dT%H:%M:%SZ')
+
+        if stream_name in ('actions', 'action_updates') and last_datetime_dt < default_date:
+            LOGGER.warning('Last datetime {} is older than 3 years, setting to default date {}'.format(last_datetime, default_date_str))
+            last_datetime = default_date_str
+
```

Comment:
Can we add a comment for `default_date` and rename it to something more relevant like `min_default_date`?

Also constraint of minimum 3 years date is specific to `actions` and `action_updates`, please mention those stream names in the warning message.

Reply:
Add a unittests for this change for scenarios.

Reply:
Added the comment but kept default_date variable name as it is to keep it simpler.
> Can we add a comment for default_date and rename it to something more relevant like min_default_date?

Added

> Also constraint of minimum 3 years date is specific to actions and action_updates, please mention those stream names in the warning message.